lib/classes/Swift/Mime/IdGenerator.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+/*
+ * This file is part of SwiftMailer.
+ * (c) 2004-2009 Chris Corbyn
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+use Egulias\EmailValidator\EmailValidator;
+
+/**
+ * Message ID generator.
+ */
+class Swift_Mime_IdGenerator implements Swift_IdGenerator
+{
+    /**
+     * @param EmailValidator $emailValidator
+     * @param string|bool    $idRight
+     */
+    public function __construct(EmailValidator $emailValidator, $idRight = false)
+    {
+        if ($idRight) {
+            $this->idRight = $idRight;
+        } else {
+            $this->idRight = !empty($_SERVER['SERVER_NAME']) ? $_SERVER['SERVER_NAME'] : 'swift.generated';
+            if (!$emailValidator->isValid('dummy@'.$this->idRight)) {
```

Comment:
Instead of depending on the emailValidator here, what about copy/pasting the regex directly here?